howto_io.md

```diff
@@ -0,0 +1,268 @@
+masking out missing
+values (if ``usemask=True``) or will fill in the missing value with the value
+specified in ``filling_values`` (default is ``np.nan`` for float, -1 for int).
+
+### Non-whitespace delimiters
+
+```
+$ cat csv.txt
+1, 2, 3
+4,, 6
+7, 8, 9
+```
+#### Masked-array output
+```
+>>> np.genfromtxt("csv.txt", delimiter=",", usemask=True)
+masked_array(
+  data=[[1.0, 2.0, 3.0],
+        [4.0, --, 6.0],
+        [7.0, 8.0, 9.0]],
+  mask=[[False, False, False],
+        [False,  True, False],
+        [False, False, False]],
+  fill_value=1e+20)
+```
+#### Array output
+```
+>>> np.genfromtxt("csv.txt", delimiter=",")
+array([[ 1.,  2.,  3.],
+       [ 4., nan,  6.],
+       [ 7.,  8.,  9.]])
+```
+#### Array output, specified fill-in value
+```
+>>> np.genfromtxt("csv.txt", delimiter=",", dtype=np.int8, filling_values=99)
+array([[ 1,  2,  3],
+       [ 4, 99,  6],
+       [ 7,  8,  9]], dtype=int8)
+```
+### Whitespace-delimited
+
+``numpy.genfromtxt`` can also parse whitespace-delimited data files
+that have missing values if
+
+ * each field has a fixed width: use the width as the `delimiter` argument
+```
+## File with width=4. The data does not have to be justified (for example, the
+## 2 in row 1), the last column can be less than width (for example, the 6 in
+## row 2), and no delimiting character is required (for instance 8888 and 9 in row 3)
+
+$cat fixedwidth.txt
+1   2      3
+44      6
+7   88889
+
+## Showing spaces as '^'
+$ tr ' ' '^' < fixedwidth.txt
+1^^^2^^^^^^3
+44^^^^^^6
+7^^^88889
+
+>>> np.genfromtxt("fixedwidth.txt",delimiter=4)
+array([[1.000e+00, 2.000e+00, 3.000e+00],
+       [4.400e+01,       nan, 6.000e+00],
+       [7.000e+00, 8.888e+03, 9.000e+00]])
+```
+ * a special value (e.g. "x") indicates a missing field: use it as the `missing_values` argument
+```
+$ cat nan.txt
+1 2 3
+44 x 6
+7  8888 9
+
+>>> np.genfromtxt("nan.txt",missing_values='x')
+array([[1.000e+00, 2.000e+00, 3.000e+00],
+       [4.400e+01,       nan, 6.000e+00],
+       [7.000e+00, 8.888e+03, 9.000e+00]])
+```
+ * you want to skip the rows with missing values: set `invalid_raise=False`
+```
+$ cat skip.txt
+1 2   3
+44    6
+7 888 9
+
+>>> np.genfromtxt("skip.txt",invalid_raise=False)
+__main__:1: ConversionWarning: Some errors were detected !
+    Line #2 (got 2 columns instead of 3)
+array([[  1.,   2.,   3.],
+       [  7., 888.,   9.]])
+```
+
+ * the delimiter whitespace character is different from the whitespace that
+   indicates missing data. For instance, if columns are delimited by `\t`,
+   then missing data will be recognized if it consists of one
+   or more spaces
+```
+$ cat tabs.txt
+1 2 3
+44    6
+7 888 9
+
+## Showing the tabs (^I) and spaces
+$ cat -T tabs.txt
+1^I2^I3
+44^I ^I6
+7^I888^I9
+
+>>> np.genfromtxt("tabs.txt",delimiter="\t",missing_values=" +")
+array([[  1.,   2.,   3.],
+       [ 44.,  nan,   6.],
+       [  7., 888.,   9.]])
+```
+
+## Read a file in .npy or .npz format

```

Comment:
Reverse the order of write and read, so you can say that "load can read files generated by any of save, savez, or savez_compressed"

Reply:
NumPy converts simpler types in other contexts, so the reader who sees this explanation may be skeptical. It's got to be more than this.  We do link to the issue, and having read it it just looks like a nontrivial problem NumPy hasn't chosen to solve.